### PR TITLE
Mask GS Regs

### LINKED ISSUE #851: Network Profile Wizard not properly rendered
To reproduce boot any game that has an online mode and it will ask you to create a network profile taking you to the network profile wizard. The game I used was NBA Ballers.

network-gui

This same bug exists on PCSX2: https://github.com/PCSX2/pcsx2/issues/287

### PR DESCRIPTION
**Pull request by Zer0xFF**
based on this fix https://github.com/PCSX2/pcsx2/pull/3279
closes #851

fixes `netgui.elf`
image

### CODE REVIEW

File: Source/gs/GSHandler.cpp [hunk 1]
```diff
@@ -604,11 +604,8 @@ void CGSHandler::WriteRegisterMassively(RegisterWriteList registerWrites, const
 
 void CGSHandler::WriteRegisterImpl(uint8 nRegister, uint64 nData)
 {
-	assert(nRegister < REGISTER_MAX);
-	if(nRegister < REGISTER_MAX)
-	{
-		m_nReg[nRegister] = nData;
-	}
+	nRegister &= REGISTER_MAX -1;
```
Comment (jpd002) [inline Source/gs/GSHandler.cpp]: Sorry, ultra nitpick. Can you place a space between the `-` and the `1`?

**Comment by jpd002** [comment]
Oh, wow, that was it? :smile: Glad we can put that one behind our backs now!

File: Source/gs/GSHandler.cpp (hunk 1 above)
Comment (Zer0xFF) [reply Source/gs/GSHandler.cpp]: done

**Comment by Zer0xFF** [comment]
I wonder if that could also be an issue for other regs?

**Comment by jpd002** [comment]
You mean, missing masks for other hardware registers? Most likely.

Thanks for this!

**Comment by Zer0xFF** [comment]
> You mean, missing masks for other hardware registers? Most likely.

yup.
Sure thing :)

**Pull request merged**